### ISSUE TITLE
Prettier doesn't format vue bindings when using type assertions

### ISSUE THREAD

**Issue by so1ve** [prettier/prettier#14918]
**Prettier 2.8.8**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8AOlAARmoCCZSAlpFALzEh5l4DOHUAnhWVZkA9CSiphmXAThiQAGhAQsMOtE7JQeAE7aIAdwAKOhBpR4c+vLw2KARtrxgA1nBgBlPBjgAZOlDhkADMLTjh7Rxc3dywnfwBzZBhtAFdwkDgMOzgAExzcnzwoeJS8eLgAMQhtDAJVYuQ2FJgIBRAACxgMHAB1drp4TliwOHdTAboANwHeRrBOWxB-MO0YQ0d42uDQ9IArTgAPdwScOABFFIh4bZwwxVjtFcbJtLasbX8YHrocmHbkAAcAAZ7nowj1HFhGu84CtJoFFABHS7wdbKMxsTgAWgCuVybW0cGRdEJ6zKWyQIVu6TCGDoSVSNJOcCoMGSdDszTghjg2j8ARudxAnGZFyugUpO0UMDwdm+v3+SAATNLHHQcAkAMIQDAUjKcACsbRSYQAKrKzFShS84ABJKD5WDuMAfFRUB3uGC8U6CuAAXz9QA)

```sh
# Options (if any): No
```

**Input:**

```vue
<template>
  <A :icon="a as any   " />
</template>
```

**Output:**

```vue
<template>
  <A :icon="a as any   " />
</template>
```

**Expected behavior:**

```vue
<template>
  <A :icon="a as any" />
  <!-- If possible I suggest format it to (a as any) since volar has some issues highlighting these code -->
</template>
```

**Comment by so1ve** [comment]
I guess it has been fixed in #14506

**Comment by so1ve** [comment]
Not actually

**Comment by fisker** [comment]
This has been fixed by #14506, a `script[lang="ts"]` is required to format it.

**Prettier 3.0.0**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8AOlAARmoCCZSAlpFALzEh5l4DOHUAnhWVZkA9CSiphmXAThjUnMACc6WGGXxQA5ixAxOrQhIXLVhEABoQEVXWidkoPIsUQA7gAUnCeyjw5XeLz2lgBGinhgANZwMADKeBhwADJ0UHDIAGZ+nHCh4VExsVgRqZrIMIoArrkgORh05VU1nKU4cACKlRDwmdk1AFacAB6xrR1dPUhZODmWAI4TcO4uWD5snAC0aXAAJrsWuuF0OKUAwhAYGHjIIHCcAKwHLVptVDAVdCGV8O5wiilpXozGoACxgGBwAHUQXR4JximA4LFvLC6AA3WG8G5gTjBEBo6oASSge1gsSUKhgVBJsRgvDaQNmICwLhykPCWBuLLufzR6UsqRyihgyzwmiujJqxUUQpuBP5zOUsEhdB2MBByAAHAAGSyKOALOj60Xi65TPqWGB4EIqtUapAAJkslRyABVrT5pky4BgQrs9jskngtJUxXAAGIQRRXd6lG54b4QEAAX2TQA)

```sh
--parser vue
--trailing-comma es5
```

**Input:**

```vue
<template>
  <A :icon="a as any   " />
</template>
<script lang="ts"></script>
```

**Output:**

```vue
<template>
  <A :icon="a as any" />
</template>
<script lang="ts"></script>

```
